cpp/src/arrow/acero/bloom_filter_test.cc

```diff
@@ -507,7 +507,11 @@ TEST(BloomFilter, Scaling) {
   num_build.push_back(4000000);
 
   std::vector<BloomFilterBuildStrategy> strategy;
+#ifndef ARROW_ENABLE_THREADING
+  strategy.push_back(BloomFilterBuildStrategy::SINGLE_THREADED);
+#else
   strategy.push_back(BloomFilterBuildStrategy::PARALLEL);
+#endif
```

Comment:
Could you use `#ifdef ... #else ... #endif` not `#ifndef ... #else ... #endif` as much as possible? A `#ifndef` (invert) and `#else` (invert) pair reduces readability.

```cpp
#ifdef ARROW_ENABLE_THREADING
  strategy.push_back(BloomFilterBuildStrategy::PARALLEL);
#else
  strategy.push_back(BloomFilterBuildStrategy::SINGLE_THREADED);
#endif
```